runelite-client/src/main/java/net/runelite/client/plugins/fps/FpsOverlay.java

```diff
@@ -0,0 +1,138 @@
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package net.runelite.client.plugins.fps;
+
+import java.awt.Color;
+import java.awt.Dimension;
+import java.awt.Graphics2D;
+import net.runelite.api.Client;
+import net.runelite.api.Constants;
+import net.runelite.api.Point;
+import net.runelite.api.events.FocusChanged;
+import net.runelite.client.ui.overlay.Overlay;
+import net.runelite.client.ui.overlay.OverlayLayer;
+import net.runelite.client.ui.overlay.OverlayPosition;
+import net.runelite.client.ui.overlay.OverlayPriority;
+import net.runelite.client.ui.overlay.OverlayUtil;
+import javax.inject.Inject;
+
+/**
+ * The built in FPS overlay has a few problems that this one does not have, most of all: it is distracting.
+ * 1. The built in one also shows memory, which constantly fluctuates and garbage collects.
+ * It is useful for devs profiling memory.
+ * 2. The built in one shifts around constantly because it is not monospace.
+ * This locks "FPS:" into one position (the far top right corner of the canvas),
+ * along with a locked position for the FPS value.
+ */
+public class FpsOverlay extends Overlay
+{
+	private static final int MAX_FPS = 50;
+	private static final int FPS_SIZE = MAX_FPS + 1;
+	private static final int Y_OFFSET = 14;
+	private static final int VALUE_X_OFFSET = 15;
+
+	// Cache of FPS number strings from 00-50
+	private final String[] fpsNums;
+
+	// Local dependencies
+	private final FpsConfig config;
+	private final Client client;
+
+	// Cached behavior from the config
+	private boolean enabled;
+	private boolean alwaysEnabled = false;
+	private boolean unfocusedEnabled = false;
+
+	// Often changing values
+	private boolean isFocused = true;
+	private Color fpsValueColor = Color.yellow;
+
+	@Inject
+	private FpsOverlay(FpsConfig config, Client client)
+	{
+		this.config = config;
+		this.client = client;
+		setLayer(OverlayLayer.ABOVE_WIDGETS);
+		setPriority(OverlayPriority.HIGH);
+		setPosition(OverlayPosition.DYNAMIC);
+
+		// Populate pre-allocated strings of FPS, these are constant and there's no reason
+		// to create additional garbage
+		// FPS should never exceed 50, we have 0-50 (51 entries)
+		String[] fpsNums = new String[FPS_SIZE];
+		for (int i = 0; i < FPS_SIZE; i++)
+		{
+			fpsNums[i] = String.format("%02d", i);
+		}
+		this.fpsNums = fpsNums;
+
+		reloadConfig();
+	}
+
+	void onFocusChanged(FocusChanged event)
+	{
+		isFocused = event.isFocused();
+		changeColor();
+	}
+
+	void reloadConfig()
+	{
+		enabled = config.drawFps();
+		alwaysEnabled = config.alwaysLimitFps();
+		unfocusedEnabled = config.unfocusedLimitFps();
+		changeColor();
+	}
+
+	private void changeColor()
+	{
+		if (alwaysEnabled || (unfocusedEnabled && !isFocused))
```

Comment:
I think this colorization is a bit confusing, you should probably color it red when it's like far off from desired FPS target, and then maybe yellow if in between and white if at target